src/redux/actionCreators.js

```diff
@@ -69,7 +69,60 @@ export function changeSource(source) {
 //TODO: implement the stop action creator synchronously
 export function stop() {
   return {
-    type: RUN,
-    payload: false
+    type: actType.STOP_RUN,
+  };
+}
+
+export function connectGoogleDrive() {
+  return dispatch => {
+    dispatch({type: actType.START_CONNECT_DRIVE});
+    var promise = new Promise((resolve, reject) => {
+      var googleDrive = "Google Drive";
+      window.setTimeout(() => resolve(googleDrive), 1000);
+      reject(new Error("Google Drive not connected"));
```

Comment:
It seems kind of strange that these all "fail" and then "succeed" a second later.